ldms/src/sampler/json/json_stream_sampler.c

```diff
@@ -379,14 +379,28 @@ int JSON_LIST_VALUE_setter(ldms_set_t set, ldms_mval_t list_mval,
 							  LDMS_V_CHAR_ARRAY, 255);
 			break;
 		case JSON_DICT_VALUE:
+			rec_idx = -1;
 			if (!rec_type_name) {
 				rc = asprintf(&rec_type_name, "%s_record", (char *)ctxt);
```

Comment:
@tom95858 This patch covers the case that there is an error in `asprintf` (`rc < 0`) here. This case was not handled before this patch.